wallet/how-to/use-non-evm-networks/starknet/connect-to-starknet.md

```diff
@@ -0,0 +1,436 @@
+<div class="row">
+  <div class="column">
+    <img src={require("../../../assets/starknet-wallet-modal.png").default} alt="Starknet wallet modal" width="360" style={{border: "1px solid #DCDCDC"}} />
+  </div>
+  <div class="column">
+    <img src={require("../../../assets/starknet-metamask-connection.png").default} alt="Starknet MetaMask connection request" width="360" style={{border: "1px solid #DCDCDC"}} />
+  </div>
+</div>
+
+After the user connects to Starknet, the dapp displays the user's connected wallet and wallet address:
+
+<p align="center">
+  <img src={require("../../../assets/starknet-dapp-connected.png").default} alt="Connected Starknet dapp" width="850" style={{border: "1px solid #DCDCDC"}} />
+</p>
+
+## Connect using `wallet_invokeSnap`
+
+Alternatively, you can manage the Snap invocation manually.
+Use the [`wallet_invokeSnap`](/snaps/reference/wallet-api-for-snaps/#wallet_invokesnap) JSON-RPC
+method to directly interact with the Starknet Snap. 
+
+:::note
+
+This option can include the use of the [Starknet Snap companion dapp](https://snaps.consensys.io/starknet), which demonstrates how to use `wallet_invokeSnap` through its [UI example](https://github.com/Consensys/starknet-snap?tab=readme-ov-file).
+
+:::
+
+### 1. Connect to the Snap
+
+Create a `src/utils` directory, and add a new file named `snapHelper.js` to the directory.
+In `snapHelper.js`, add a `connect` function and a `callSnap` helper function as follows.
+This file handles the interactions with the Starknet Snap:
+
+```javascript title="snapHelper.js"
+const snapId = "npm:starknet-snap";
+
+export async function connect() {
+  await ethereum.request({
+    method: "wallet_requestSnaps",
+    params: {
+      [snapId]: {},
+    },
+  });
+}
+
+export async function callSnap(method, params) {
+  try {
+    const response = await ethereum.request({
+      method: "wallet_invokeSnap",
+      params: {
+        snapId,
+        request: {
+          method,
+          params,
+        },
+      },
+    });
+    console.log(`${method} response:`, response);
+    return response;
+  } catch (err) {
+    console.error(`${method} error:`, err);
+    alert(`${method} error: ${err.message || err}`);
+    throw err;
+  }
+}
+```
+
+:::note
+
+To connect to Starknet, the dapp user must add the Starknet Snap to MetaMask.
+Make sure to [handle user rejections](troubleshoot.md#handle-user-rejection).
+
+:::
+
+### 2. Call a specific Snap method
+
+Use the `callSnap` function to call a specific Snap method.
+The following example calls [`starkNet_createAccount`](../../../reference/non-evm-apis/starknet-snap-api.md#starknet_createaccount):
+
+```javascript
+const deploy = false; // Set to true to deploy the actual account.
+const addressIndex = 0; // Specify which address to derive.
+const chainId = "0x534e5f5345504f4c4941"; // Chain ID of the network to use.
+
+const accountInfo = await callSnap("starkNet_createAccount", { addressIndex, deploy, chainId });
+```
+
+### Examples
```

Comment:
I think it might be helpful to work these docs in phases/ stages. I can add that as a phase II activity in addition to linking to the support docs (as well as  whatever tasks may carry over from the MVP).